File: app/src/main/java/org/mifos/mobilebanking/utils/NotificationComparator.java
package org.mifos.mobilebanking.utils;

import org.mifos.mobilebanking.models.notification.MifosNotification;

import java.util.Comparator;

/**
 * Created by dilpreet on 14/9/17.
 */

public class NotificationComparator implements Comparator<MifosNotification> {
    @Override
    public int compare(MifosNotification mifosNotification1, MifosNotification mifosNotification2) {
        return mifosNotification2.getTimeStamp() < mifosNotification1.getTimeStamp() ? -1 :
                mifosNotification1.getTimeStamp() > mifosNotification2.getTimeStamp() ? 1 : 0;
    }
}

File: app/src/main/java/org/mifos/mobile/utils/NotificationComparator.kt
package org.mifos.mobile.utils

import org.mifos.mobile.models.notification.MifosNotification
import java.util.*

/**
 * Created by dilpreet on 14/9/17.
 */
class NotificationComparator : Comparator<MifosNotification> {
    override fun compare(firstNotification: MifosNotification, secondNotification: MifosNotification): Int {
        return when {
            // comparator function logic to sort notifications in the descending order of their timeStamp :
            firstNotification.timeStamp > secondNotification.timeStamp -> -1
            firstNotification.timeStamp < secondNotification.timeStamp -> 1
            else -> 0
        }
    }
}
